### PR TITLE
feat!: Make Summary Box more flexible

### LINKED ISSUE #1916: [!feat] Make summary box components more flexible
**Does your feature request relate to a specific USWDS component?**

[Summary Box](https://designsystem.digital.gov/components/summary-box/)

**Is your feature request related to a problem? Please describe.**

Right now it is not possible to compose a summary box flexibly, IE, if someone wanted to place a logo above the header within the summary box. 
Additionally, this should include in scope the ability to specify the heading level on a summary box (currently uses the default of `h3`, but like many components containing headers, the heading level is important to be customizable)

**Describe the solution you'd like**

Build SummaryBoxHeading and SummaryBoxText components to allow consumers to compose a summary box themselves.

> ⚠️ Note: This is likely a breaking change as it would alter the existing summary box api

**Describe alternatives you've considered**

It would be possible to let the current implementation live in parallel with a more flexible implementation, but I don't think we want to be maintaining multiple versions of the same components or extensive conditional rendering logic within a component for the sake of backwards compatibility alone

**Additional context**

Lets say I wanted to do something like this:
image

### PR DESCRIPTION
**Pull request by Igarfinkle**
> ### ⚠️ Set the following as the commit body when merging this PR: `BREAKING CHANGE: SummaryBox now exposes sub-components (SummaryBoxHeading and SummaryBoxContent) for a more compositional API. Consumers will need to update their implementation to match.`

# Summary

Makes the SummaryBox component more flexible, separating out SummaryBoxHeading and SummaryBoxHeading into their own components.
Is NOT backwards compatible. 

## Related Issues or PRs

Closes #1916 

## How To Test

Try the SummaryBox Storybook story to ensure it still displays properly

### CODE REVIEW

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (kimallen) [inline src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx:13]: You could give `headingLevel` a default here rather than handle that below with `headingLevel = h3`.  That might be more explicit for someone reading this code.

File: src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx:58]: This test _was_ (although the name didn't reflect it explicitly) the only test that validated that a heading as being rendered. Perhaps `testSummaryBoxContent` should be updated to include the heading now?

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx:5]: instead of a heading prop, can we use `children` here? That way the api can look like `<SummaryBoxHeading>My Heading Content</SummaryBoxHeading>`

That'll also allow more flexibility over just `string` in case someone wanted to otherwise stylize or pass in custom tags.

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx:13]: @suzubara @haworku @kylehilltruss What are your thoughts on having a default value here? In the past we've done it as a way to preserve backwards compatibility (non-breaking changes) where we've introduced the ability to customize the heading level. 

In this case, this is a breaking change anyways since the SummaryBox API is changing to be more compositional in nature. Should we just rip off the bandaid and make it a mandatory prop? IMO I think we should, since there's no real significance to `h3` other than it is what the USWDS example happens to use.

**Comment by kimallen** [comment]
I'm wondering if it would be worth breaking out the `headingLevel` change into its own issue/PR from the breaking change so that the `headingLevel` flexibility can be incorporated as a minor non-breaking change before the breaking change.

**Review by brandonlenz — COMMENTED** [review]
@Igarfinkle thanks so much for picking this up!

A couple of suggestions and one question to some other maintainers in my review here. 

One nit: We don't seem to have a unified pattern for how we name folders when double nesting things, (e.g. in this case `SummaryBox/SummaryBox`, or `Footer/Footer` or `card/Card`). JS, React, and TS are pretty unopinionated about this, so none are wrong.

Personally, if a folder does not directly contain a component file, I think it should be lower-cased (ie. `summarybox/SummaryBox`) 

@kylehilltruss @suzubara @haworku thoughts? Some people prefer snakeCase or kebab-case for multi-word directories. If we come to a consensus, I'll draft a PR + ADR to set them all straight 😉

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (brandonlenz) [reply src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx:13]: @kimallen also makes a good point [here](https://github.com/trussworks/react-uswds/pull/1929#issuecomment-1081003074) about multi-staging this process.

File: src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx
Comment (Igarfinkle) [reply src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx:58]: I don't think that's needed: now the heading is just... part of the children of SummaryBox.

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (Igarfinkle) [reply src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx:13]: As long as we're not taking the USWDS example as the source of truth, then yes, I think we should remove the default value

File: src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx
Comment (brandonlenz) [reply src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx:58]: Right, but shouldn't the tests somewhat reflect how we expect the component to be used? In which case, the `render` should include those tags with `testSummaryBoxContent` rendered inside the `SummaryBoxContent`.

And then I think there's still value in checking for the header to be rendered within. What we could probably do here is combine the second test (renders passed in children) with the first test, make sure we do a check for the header, and call it a day

File: src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx:73]: I see you've updated `testSummaryBoxContent` to include the header and content tags. Can we put the check for the header here to assuage my concerns?

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (haworku) [reply src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx]: Yes to removing default value-  this is a breaking change we should put things in the ideal state we would like it to be in going forward.

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx:20]: nit: I prefer to save the data-testid for when something really can't easily be queried from the dom by conventional means.

Before this PR the test used the following: `expect(getByRole('heading')).toBeInTheDocument()`. I think it could also work with `getByText`, knowing what the heading's content is.

Comment (Igarfinkle) [reply src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx]: ah, I forgot about getByRole!

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.test.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.test.tsx:2]: Let's get rid of this unused import

File: src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx
Comment (brandonlenz) [inline src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx:3]: This one also

**Review by brandonlenz — APPROVED** [review]
Approving but with a couple of tiny things to fix listed below.

We'll likely want to time the release of this (since it's breaking), so lets work that out in our slack

File: src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx
Comment (kimallen) [reply src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx]: @Igarfinkle Did you decide to use `heading` prop anyway instead of `children`? I liked @brandonlenz 's suggestion

Comment (kimallen) [reply src/components/SummaryBox/SummaryBoxHeading/SummaryBoxHeading.tsx]: either way, a comment here to explain why it was resolved without changes would be helpful to future viewers.

File: src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx
Comment (Igarfinkle) [reply src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx]: Done. I'm surprised eslint doesn't find this problematic enough to fail the CI check.

Comment (haworku) [reply src/components/SummaryBox/SummaryBox/SummaryBox.test.tsx]: Good point. I think that unused direction might be a warning not error? Perhaps we need to adjust to have CI checks fail on warnings too - I think its just adding `--max-warnings=0` to our eslint call.

**Review by haworku — APPROVED** [review]
LGTM. I made https://github.com/trussworks/react-uswds/issues/2000 to address the issue brandon brought up.

**Review by brandonlenz — APPROVED** [review]
Approving now since I think we've done all the communicating around an intentionally breaking release

**Comment by brandonlenz** [comment]
@Igarfinkle when merging this, please make sure that the commit body of the merge commit contains some verbiage like 

```
BREAKING CHANGE: SummaryBox now exposes sub-components (SummaryBoxHeading and SummaryBoxContent) for a more compositional API. Consumers will need to update their implementation to match.
```

**Pull request merged**
